efwyugrbdjknlfgk

In [2]:
library(repr)
library(tidyverse)
library(tidymodels)
options(repr.matrix.max.rows = 6)
source('tests.R')
source("cleanup.R")

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()

── Attaching packages ────────────────────────────────────── tidymodels 1.0.0 ──

✔ broom        1.0.0     ✔ rsample      1.0.0
✔ dials        1.0.0     ✔ tune         1.0.0
✔ infer        1.0.2     ✔ workflows    1.0.0
✔ modeldata    1.0.0     ✔ workflowsets 1.0.0
✔ parsnip      1.0.0     ✔ yardstick    1.0.0
✔ recipes      1.0.1     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filter()   masks stats::filter()
✖ recipes::fixed()  masks stringr::fixed()
✖ dplyr::lag()      masks stats::lag()
✖ yardstick::spec() masks r

ERROR: Error in file(filename, "r", encoding = encoding): cannot open the connection


In [40]:
data<-read_csv("processed.cleveland.data",col_names=FALSE)
data
data<-data|>
    mutate(X2=as.factor(X2),X3=as.factor(X3),X6=as.factor(X6),X7=as.factor(X7),X9=as.factor(X9),X11=as.factor(X11),
             X12=as.factor(X12),X13=as.factor(X13),X14=as.factor(X14))

names<-c("age","sex","cp","trestbps","chol","fbs","restecg","thalach","exang","oldpeak","slope","ca","thal","num")
colnames(data)<-make.names(names)
data

Rows: 303 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (2): X12, X13
dbl (12): X1, X2, X3, X4, X5, X6, X7, X8, X9, X10, X11, X14

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


X1 X2 X3 X4  X5  X6 X7 X8  X9 X10 X11 X12 X13 X14
1   63 1  1  145 233 1  2  150 0  2.3 3   0.0 6.0 0  
2   67 1  4  160 286 0  2  108 1  1.5 2   3.0 3.0 2  
3   67 1  4  120 229 0  2  129 1  2.6 2   2.0 7.0 1  
⋮   ⋮  ⋮  ⋮  ⋮   ⋮   ⋮  ⋮  ⋮   ⋮  ⋮   ⋮   ⋮   ⋮   ⋮  
301 57 1  4  130 131 0  0  115 1  1.2 2   1.0 7.0 3  
302 57 0  2  130 236 0  2  174 0  0.0 2   1.0 3.0 1  
303 38 1  3  138 175 0  0  173 0  0.0 1   ?   3.0 0

age sex cp trestbps chol fbs restecg thalach exang oldpeak slope ca  thal
1   63  1   1  145      233  1   2       150     0     2.3     3     0.0 6.0 
2   67  1   4  160      286  0   2       108     1     1.5     2     3.0 3.0 
3   67  1   4  120      229  0   2       129     1     2.6     2     2.0 7.0 
⋮   ⋮   ⋮   ⋮  ⋮        ⋮    ⋮   ⋮       ⋮       ⋮     ⋮       ⋮     ⋮   ⋮   
301 57  1   4  130      131  0   0       115     1     1.2     2     1.0 7.0 
302 57  0   2  130      236  0   2       174     0     0.0     2     1.0 3.0 
303 38  1   3  138      175  0   0       173     0     0.0     1     ?   3.0 
    num
1   0  
2   2  
3   1  
⋮   ⋮  
301 3  
302 1  
303 0

X1. #3 (age)

X2. #4 (sex)\
(1 = male; 0 = female)

X3. #9 (cp)\
(chest pain type)
-- Value 1: typical angina
-- Value 2: atypical angina
-- Value 3: non-anginal pain
-- Value 4: asymptomatic

X4. #10 (trestbps)\
resting blood pressure (in mm Hg on admission to the hospital)

X5. #12 (chol)\
serum cholestoral in mg/dl

X6. #16 (fbs)\
(fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)

X7. #19 (restecg)\
 resting electrocardiographic results
-- Value 0: normal
-- Value 1: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV)
-- Value 2: showing probable or definite left ventricular hypertrophy by Estes' criteria

X8. #32 (thalach)\
maximum heart rate achieved

X9. #38 (exang)\
exercise induced angina (1 = yes; 0 = no)

X10. #40 (oldpeak)\
ST depression induced by exercise relative to rest

X11. #41 (slope)\
the slope of the peak exercise ST segment
-- Value 1: upsloping
-- Value 2: flat
-- Value 3: downsloping

X12. #44 (ca)\
number of major vessels (0-3) colored by flourosopy

X13. #51 (thal)\
3 = normal; 6 = fixed defect; 7 = reversable defect

X14. #58 (num) (the predicted attribute)\
diagnosis of heart disease (angiographic disease status)
-- Value 0: < 50% diameter narrowing
-- Value 1: > 50% diameter narrowing
(in any major vessel: attributes 59 through 68 are vessels)